### PR TITLE
Added ability to pass runsettings as commandline parameters after --

### PR DESCRIPTION
**Pull request by harshjain2**
- Added ability to pass runsettings arguments after -- for dotnet test.
- Updated documentation of --logger.

### CODE REVIEW

File: src/dotnet/commands/dotnet-test/LocalizableStrings.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ internal class LocalizableStrings
         public const string CmdLoggerOption = "LoggerUri/FriendlyName";
 
         public const string CmdLoggerDescription = @"Specify a logger for test results. 
-                                        Example: --logger:trx";
+                                        Example: --logger ""trx[;LogFileName=<Defaults to unique file name>]""";
```
Comment (livarcocc) [inline src/dotnet/commands/dotnet-test/LocalizableStrings.cs]: @srivatsn More string updates. Ok to take them?

File: src/dotnet/commands/dotnet-test/LocalizableStrings.cs [hunk 2]
```diff
@@ -52,5 +52,7 @@ internal class LocalizableStrings
                                         Logs are written to the provided file.";
 
         public const string CmdNoBuildDescription = @"Do not build project before testing.";
+
+        public const string RunSettingsArgsHelpText = @"Any extra options that should be passed to MSBuild or adapter. See 'dotnet msbuild -h' and 'dotnet vstest --help' for available options.";
```
Comment (livarcocc) [inline src/dotnet/commands/dotnet-test/LocalizableStrings.cs]: @srivatsn And here.

Comment (srivatsn) [inline src/dotnet/commands/dotnet-test/LocalizableStrings.cs]: Yes. @nguerrera

**Comment by harshjain2** [comment]
cc @blackdwarf @codito

File: src/dotnet/commands/dotnet-test/Program.cs [hunk 3]
```diff
@@ -165,6 +166,13 @@ public static int Run(string[] args)
...
 
+                // Get runsetings options specified after -- 
+                if (cmd.RemainingArguments != null && cmd.RemainingArguments.Count > 0)
+                {
+                    var runSettingsOptions = GetRunSettingsOptions(cmd.RemainingArguments);
+                    msbuildArgs.Add(string.Format("/p:VSTestCLIRunSettings=\"{0}\"", string.Join(";", runSettingsOptions)));
+                }
+
                 // Add remaining arguments that the parser did not understand,
                 msbuildArgs.AddRange(cmd.RemainingArguments);
```
Comment (Faizan2304) [inline src/dotnet/commands/dotnet-test/Program.cs]: Do we need this line of code??

File: src/dotnet/commands/dotnet-test/LocalizableStrings.cs (hunk 2 above)
Comment (Faizan2304) [inline src/dotnet/commands/dotnet-test/LocalizableStrings.cs]: The string doesn't seems correct. This argument is not for adapter but for commandline runsttings argument of vstest.
It should be some thing like "Any extra options that will be consider as commndline runsettings argument or will pass to msbuild. See 'dotnet msbuild -h' and 'dotnet vstest --help' for available options.
Example: dotnet test TestProject.csproj -- RunConfiguration.ResultsDirectory=ResultsDirectoryName"

Comment (harshjain2) [inline src/dotnet/commands/dotnet-test/LocalizableStrings.cs]: @blackdwarf Please suggest the documentation text for dotnet test. 

Here is how a dotnet command would look like :
dotnet test foo.csproj -- logger trx {any argument to pass to msbuild} -- {run settings arguments to pass to vstest.console}
E.g.:  dotnet test foo.csproj --logger trx /Verbosity:detailed -- RunConfiguration.ResultsDirectory="C:\users\user\desktop\ResultsDirectory"

Comment (jonsequitur) [inline src/dotnet/commands/dotnet-test/LocalizableStrings.cs]: I was surprised that a different help switch appears to be needed for these two commands. Is this by design? @livarcocc @piotrpMSFT

Comment (jonsequitur) [inline src/dotnet/commands/dotnet-test/LocalizableStrings.cs]: @blackdwarf

File: src/dotnet/commands/dotnet-test/LocalizableStrings.cs (hunk 1 above)
Comment (piotrpMSFT) [inline src/dotnet/commands/dotnet-test/LocalizableStrings.cs]: I'd prefer this be a separate PR. If we get another loc chance then we'll merge, otherwise it's a nice-to-have.

File: src/dotnet/commands/dotnet-test/LocalizableStrings.cs [hunk 4]
```diff
@@ -52,5 +52,8 @@ internal class LocalizableStrings
                                         Logs are written to the provided file.";
 
         public const string CmdNoBuildDescription = @"Do not build project before testing.";
+
+        public const string RunSettingsArgsHelpText = @"Any extra commandline runsettings arguments that should be passed to vstest. See 'dotnet vstest --help' for available options.
+                                        Example: -- RunConfiguration.ResultsDirectory=""C:\users\user\desktop\Results Directory"" MSTest.DeploymentEnabled=false";
```
Comment (piotrpMSFT) [inline src/dotnet/commands/dotnet-test/LocalizableStrings.cs]: This also seems nice-to-have. It's quite advanced of functionality and will be nicely documented by e.g. xunit folks when they help their customers use this feature. 

@nguerrera how are we doing with loc handoffs?

File: src/dotnet/commands/dotnet-test/Program.cs [hunk 5]
```diff
@@ -165,6 +165,13 @@ public static int Run(string[] args)
                     msbuildArgs.Add(argRoot.Value);
                 }
 
+                // Get runsetings options specified after -- 
+                if (cmd.RemainingArguments != null && cmd.RemainingArguments.Count > 0)
+                {
+                    var runSettingsOptions = GetRunSettingsOptions(cmd.RemainingArguments);
+                    msbuildArgs.Add(string.Format("/p:VSTestCLIRunSettings=\"{0}\"", string.Join(";", runSettingsOptions)));
```
Comment (piotrpMSFT) [inline src/dotnet/commands/dotnet-test/Program.cs]: Why do we use `GetSemiColonEscapedArgs` above but not here?

File: test/dotnet-test.Tests/GivenDotnetTestBuildsAndRunsTestfromCsproj.cs [hunk 6]
```diff
@@ -140,6 +140,47 @@ public void TestWillCreateTrxLogger()
             }
         }
 
+        [Fact]
```
Comment (piotrpMSFT) [inline test/dotnet-test.Tests/GivenDotnetTestBuildsAndRunsTestfromCsproj.cs]: We also need a test for the custom RunSettingsOptions...

Comment (harshjain2) [inline test/dotnet-test.Tests/GivenDotnetTestBuildsAndRunsTestfromCsproj.cs]: This test is testing both the features : a.) newly added log file name. b.) custom RunSettingsOptions. 
`--logger \"trx;logfilename=custom.trx\" -- RunConfiguration.ResultsDirectory=" + trxLoggerDirectory`

**Comment by nguerrera** [comment]
> @nguerrera how are we doing with loc handoffs?

@piotrpMSFT I'm arranging for another one. Please get this (and any other string changes) + xlf conversion merged by EOD. Please still avoid large changes as the word count affects the turn around time from the loc team and we need this one back ASAP.

File: src/dotnet/commands/dotnet-test/Program.cs (hunk 5 above)
Comment (harshjain2) [inline src/dotnet/commands/dotnet-test/Program.cs]: GetSemiColonEscapeArgs is just replacing semi colon with %3b in all the elements.
GetRunSettingsOptions is extracting the run settings after '--' and replacing semi colon with %3b as well.
